src/network-proxy.ts

```diff
@@ -0,0 +1,88 @@
+import * as shell from "shelljs";
+import path from 'path';
+
+const networkSetup = '/usr/sbin/networksetup';
+
+export function isProxyAvaliable() {
+  const proxy = process.env.HTTP_PROXY;
+  return proxy && Array.isArray(proxy.split(':')) && proxy.split(':').length > 2;
+}
+
+function getProxySetting() {
+  const proxy = process.env.HTTP_PROXY?.split(':') || [];
+  if (proxy?.length < 2) {
+    return;
+  }
+  return {
+    proxyHost: proxy[1].replaceAll('/', ''),
+    proxyPort: proxy[2],
+  }
+}
+
+function findNetworkServiceOnMac() {
+  const networkInfo = shell.exec(`${networkSetup} -listnetworkserviceorder`, {async: false}).stdout;
+  const lines = networkInfo.split('\n');
+  for (let i = 0; i < lines.length; i++) {
+    const line = lines[i];
+    if (line.includes('Device: en')) {
+      // Network service name would be shown like: (1) Wi-Fi.
+      // Extract "Wi-Fi" from this line.
+      const service = lines[i-1].substring(4);
+      const serviceInfo = shell.exec(`${networkSetup} -getinfo "${service}"`, {async: false}).stdout;
+      for (const l of serviceInfo.split('\n')) {
+        if (l.includes('IP address') && !l.includes('IPv6') && !l.includes('none')) {
+          return service;
+        }
+      }
+    }
+  }
+  return 'Ethernet';
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function setupMacProxy(proxy: any) {
+  const {
+    proxyHost,
+    proxyPort,
+  } = proxy;
+  const networkService = findNetworkServiceOnMac();
+  shell.exec(`sudo ${networkSetup} -setwebproxy "${networkService}" ${proxyHost} ${proxyPort}`, {async: false});
+  shell.exec(`sudo ${networkSetup} -setsecurewebproxy "${networkService}" ${proxyHost} ${proxyPort}`, {async: false});
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function setupWinProxy(proxy: any) {
```

Comment:
Why proxy type any (techdebt)? You know the type after all, you created the object in `getProxySetting()`.